src/Splat/Disposables/CompositeDisposable.cs

```diff
@@ -31,7 +31,10 @@ internal sealed class CompositeDisposable : IDisposable
     /// <exception cref="ArgumentOutOfRangeException"><paramref name="capacity"/> is less than zero.</exception>
     public CompositeDisposable(int capacity)
     {
-        capacity.ThrowArgumentNullExceptionIfNull(nameof(capacity));
+        if (capacity < 0)
+        {
+            throw new ArgumentOutOfRangeException(nameof(capacity), "Capacity cannot be negative.");
```

Comment:
ArgumentOutOfRangeException has a throw helper now which is more efficient then the branch here for checking if a value is 0 or below, or negative etc.

Reply:
I believe there is only a ThrowArgumentNullExceptionIfNull, not a ThrowArgumentOutOfRangeException